File: api/Utilities/PagedList.cs
namespace api.Utilities;
public class PagedList<T>(int page, int size, int totalCount, T[] items)
{
    public int Page { get; private set; } = page;
    public int Size { get; private set; } = size;
    public int TotalCount { get; private set; } = totalCount;
    public int TotalPages => (int)Math.Ceiling(TotalCount / (double)Size);
    public bool HasPreviousPage => Page > 1;
    public bool HasNextPage => Page < Size;
    public T[] Items { get; private set; } = items;
}

File: client/src/types/pagedList.ts
export class PagedList<T> {
  page: number;
  size: number;
  totalCount: number;
  items: T[];

  constructor(page: number, size: number, totalCount: number, items: T[]) {
    this.page = page;
    this.size = size;
    this.totalCount = totalCount;
    this.items = items;
  }

  get totalPages(): number {
    return Math.ceil(this.totalCount / this.size);
  }

  get hasPreviousPage(): boolean {
    return this.page > 1;
  }

  get hasNextPage(): boolean {
    return this.page < this.totalPages;
  }
}
